### PR TITLE
chore: fix typos

### PR DESCRIPTION
**Pull request by hoangdv2429**
## Description

This PR will fix every typos exist on current repo. After this we can add another CI to check for typo per PR submit.

#### PR checklist

- [ ] Tests written/updated
- [ ] Changelog entry added in `.changelog` (we use
  [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [x] Updated relevant documentation (`docs/` or `spec/`) and code comments

### CODE REVIEW

**Comment by rootulp** [comment]
> After this we can add another CI to check for typo per PR submit.

Which linter or CI check do you propose adding? This repo already enabled https://github.com/celestiaorg/celestia-core/blob/4b6c2dda7dfb4cd9c74986a2a4a98d1bab44d961/.golangci.yml#L18

File: privval/msgs_test.go [hunk 1]
```diff
@@ -71,7 +71,7 @@ func TestPrivvalVectors(t *testing.T) {
 	proposal := exampleProposal()
 	proposalpb := proposal.ToProto()
 
-	// Create a Reuseable remote error
+	// Create a Reusable remote error
```
Comment (rootulp) [inline privval/msgs_test.go:74]: [optional] IMO this shouldn't be capitalized
```suggestion
	// Create a reusable remote error
```

**Review by rootulp — APPROVED** [review]
LGTM. Some of these typos may exist in upstream https://github.com/cometbft/cometbft if you'd like to make a similar PR there

Thanks for your contribution!

File: privval/msgs_test.go (hunk 1 above)
Comment (hoangdv2429) [reply privval/msgs_test.go:74]: yeah... I'll fix that as well.

**Review by rootulp — APPROVED** [review]

**Comment by hoangdv2429** [comment]
`=== Informational ===

Found 2 vulnerabilities in packages that you import, but there are no call
stacks leading to the use of these vulnerabilities. You may not need to
take any action. See https://pkg.go.dev/golang.org/x/vuln/cmd/govulncheck
for details.

Vulnerability #1: GO-2023-2186
    Incorrect detection of reserved device names on Windows in path/filepath
  More info: https://pkg.go.dev/vuln/GO-2023-2186
  Standard library
    Found in: path/filepath@go1.21.3
    Fixed in: path/filepath@go1.21.4

Vulnerability #2: GO-2023-2102
    HTTP/2 rapid reset can cause excessive work in net/http
  More info: https://pkg.go.dev/vuln/GO-2023-2102
  Module: golang.org/x/net
    Found in: golang.org/x/net@v0.14.0
    Fixed in: golang.org/x/net@v0.17.0

Your code is affected by 2 vulnerabilities from the Go standard library.`
Yes, I agree. Seems like we need to bump those deps.

**Comment by hoangdv2429** [comment]
The failing check could be from json or toml. Please let's me check.

**Comment by hoangdv2429** [comment]
@rootulp Can you rerun the e2e test sir, I don't think I touch any config files

**Comment by rootulp** [comment]
> @rootulp Can you rerun the e2e test sir, I don't think I touch any config files

Thanks for the ping! Sorry the CI on this repo is flakey. Re-running.

**Review by staheri14 — APPROVED** [review]

**Pull request merged**